### PR TITLE
feat: context action to create overwrites

### LINKED ISSUE #34: Right click context menu to add overwrite
- As mentioned in https://github.com/GodotModding/godot-mod-loader/pull/74
- [x] Add a right click context menu "add overwrite"
- [x] Create the `/overwrites` dir, if it's missing
- [x] Mirror the game's file system inside the mods `/overwrites` dir
- [x] Copy the selected resource to the created dir
- [x] Create the *overwrites.gd* file, in the root mod dir, if it's missing
- [x] Add the required code to the `_init()` func of *overwrites.gd*

### Required Code per overwrite:
```gdscript
var overwrite_0 = preload("res://mods-unpacked/KANA-OverwritesTest/overwrites/assets/images/GodotModded.png")
overwrite_0.take_over_path("res://assets/images/GodotModded.png")
```

### Example structure:
image

### PR DESCRIPTION
**Pull request by Qubus0**
Add the ability to overwrite assets in overwrites.gd via file system context menu
closes #34

### CODE REVIEW

File: addons/mod_tool/interface/file_system/file_system_context_actions.gd [hunk 1]
```diff
@@ -183,6 +212,62 @@ static func add_script_extension_to_mod_main(file: File, extension_path: String)
...
+	file.close()
+
+	# Construct the line required to preload the asset and take over the path
+	var asset_overwrite_line := "\tpreload(%s).take_over_path(%s)\n" % [quote_string(asset_path), quote_string(vanilla_asset_path)]
+
+	# Check if that asset is already being overwritten
+	if asset_overwrite_line.strip_edges() in file_content:
+		return
+
+	# If there is no modLoader.install_script_extension yet, put it at the end of install_script_extensions
```
Comment (KANAjetzt) [inline addons/mod_tool/interface/file_system/file_system_context_actions.gd:261]: I don't get that comment why  `modLoader.install_script_extension` ?

**Review by KANAjetzt — APPROVED** [review]
nice 👌  great work ty 🥇

**Comment by KANAjetzt** [comment]
Just as reminder you may want to change the PR title to `feat:´

File: addons/mod_tool/interface/file_system/file_system_context_actions.gd (hunk 1 above)
Comment (Qubus0) [reply addons/mod_tool/interface/file_system/file_system_context_actions.gd:261]: whoops, a leftover from duplicating

**Pull request merged**